File: src/main/java/com/fliliy/secondhand/entity/Category.java
package com.fliliy.secondhand.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "categories")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Category {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    
    @Column(name = "parent_id")
    private Integer parentId = 0;
    
    @Column(nullable = false, length = 100)
    private String name;
    
    @Column(length = 500)
    private String icon;
    
    @Column(length = 500)
    private String description;
    
    @Column(name = "sort_order")
    private Integer sortOrder = 0;
    
    @Column(name = "is_active")
    private Boolean isActive = true;
    
    @Column(name = "product_count")
    private Integer productCount = 0;
    
    @Column(name = "created_at", updatable = false)
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime updatedAt;
    
    // 子分类列表（用于API响应，不映射到数据库）
    @Transient
    private List<Category> children;
    
    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: apps/mobile/lib/features/category/domain/entities/category.dart
class CategoryEntity {
  final int id;
  final String name;
  final String? icon;
  final int parentId;
  final int sortOrder;
  final bool isActive;
  final int? productCount;

  /// Children categories (non-null, default empty)
  final List<CategoryEntity> children;

  /// Optional H5 link URL
  final String? linkUrl;

  const CategoryEntity({
    required this.id,
    required this.name,
    this.icon,
    required this.parentId,
    required this.sortOrder,
    required this.isActive,
    this.productCount,
    this.children = const <CategoryEntity>[],
    this.linkUrl,
  });
}
